src/theory/rewriter.cpp

```diff
@@ -135,12 +121,15 @@ Node Rewriter::rewriteTo(theory::TheoryId theoryId, Node node) {
     if (rewriteStackTop.nextChild == 0) {
 
       // Check if the pre-rewrite has already been done (it's in the cache)
-      Node cached = Rewriter::getPreRewriteCache((TheoryId) rewriteStackTop.theoryId, rewriteStackTop.node);
+      Node cached = getPreRewriteCache((TheoryId)rewriteStackTop.theoryId,
```

Comment:
Just checked why we even need a cast here. `theoryId` is a 8-bit unsigned in struct `RewriteStackElement`. Wouldn't it be better to have a `getTheoryId` that makes the cast for you?

Reply:
Good idea, done.